frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.tsx

```diff
@@ -1,29 +1,105 @@
 /* Copyright Contributors to the Open Cluster Management project */
 
-import { AcmInlineStatus, StatusType } from '@open-cluster-management/ui-components'
+import { V1CustomResourceDefinitionCondition } from '@kubernetes/client-node/dist/gen/model/v1CustomResourceDefinitionCondition'
+import { AcmButton, AcmInlineStatus, StatusType } from '@open-cluster-management/ui-components'
 import { ButtonVariant } from '@patternfly/react-core'
 import { ArrowCircleUpIcon, ExternalLinkAltIcon } from '@patternfly/react-icons'
-import { useState } from 'react'
+import { ReactNode, useState } from 'react'
 import { useTranslation } from 'react-i18next'
 import { RbacButton } from '../../../../../components/Rbac'
-import { Cluster, ClusterStatus } from '../../../../../lib/get-cluster'
+import {
+    checkCuratorLatestFailedOperation,
+    checkCuratorLatestOperation,
+    Cluster,
+    ClusterStatus,
+    CuratorCondition,
+    getConditionStatusMessage,
+} from '../../../../../lib/get-cluster'
 import { rbacCreate, rbacPatch } from '../../../../../lib/rbac-util'
-import { ClusterCuratorDefinition } from '../../../../../resources/cluster-curator'
+import { AnsibleJob, getLatestAnsibleJob } from '../../../../../resources/ansible-job'
+import { ClusterCurator, ClusterCuratorDefinition } from '../../../../../resources/cluster-curator'
 import { BatchUpgradeModal } from './BatchUpgradeModal'
+
 export const backendUrl = `${process.env.REACT_APP_BACKEND_PATH}`
 
-export function DistributionField(props: { cluster?: Cluster }) {
+export function DistributionField(props: {
+    cluster?: Cluster
+    clusterCurator?: ClusterCurator | undefined
+    ansibleJobs?: AnsibleJob[] | undefined
+}) {
     const { t } = useTranslation(['cluster'])
     const [open, toggleOpen] = useState<boolean>(false)
     const toggle = () => toggleOpen(!open)
+    const clusterCurator = props.clusterCurator
+    const upgradeInfo = props.cluster?.distribution?.upgradeInfo
```

Comment:
all the other conditional flows just reference the full object path, so we should have consistency with everything